### PR TITLE
Wip/hughsie/fwupdtool unsafe

### PR DESCRIPTION
**Pull request by hughsie**

### CODE REVIEW

**Review by superm1 — APPROVED** [review]

File: libfwupdplugin/fu-volume.c [hunk 1]
```diff
@@ -81,7 +81,7 @@ fu_volume_get_id (FuVolume *self)
 gchar *
 fu_volume_get_mount_point (FuVolume *self)
 {
-	const gchar **mountpoints = NULL;
+	g_autofree const gchar **mountpoints = NULL;
```
Comment (superm1) [inline libfwupdplugin/fu-volume.c:84]: With this change, does it make sense to perhaps `g_steal_pointer()` on the function return?

Comment (hughsie) [reply libfwupdplugin/fu-volume.c:84]: No, as it's only returning `container` and we're returning one of the array _elements_ that is returned const.

**Comment by superm1** [comment]
For my own learning, how did you manage to narrow this down to the inotify firing?  I couldn't step into a `g_main_loop` run with a vscode frontend for gdb.

**Comment by hughsie** [comment]
> how did you manage to narrow this down to the inotify firing

Good question. Valgrind was really helpful in telling me the memory was being used after free, and I then I worked back to places that were allocating the object. And a good dollop of luck too.

**Pull request merged**